src/monsters.h

```diff
@@ -181,8 +196,8 @@ class Monsters
 
 		std::map<std::string, uint32_t> monsterNames;
 		std::map<MonsterType*, std::string> monsterScriptList;
-		std::map<uint32_t, MonsterType*> monsters;
-		LuaScriptInterface* scriptInterface;
+		std::map<uint32_t, MonsterType> monsters;
```

Comment:
Change the key to string. I don't see anything taking advantage of a uint32_t key here, we're always getting the id by looking up the name in monsterNames. You can also remove monsterNames while you're at it.